lib/actions.js

```diff
@@ -58,6 +66,64 @@ const _createAccount = (walletId, accountId, accountInfo) => ({
   type: WALLET_CREATE_ACCOUNT,
   payload: { walletId, accountId, accountInfo }
 });
+// multisig wallet helpers
+const _getMultisigWallets = (multisigWallets) => ({
+  type: SET_MULTISIG_WALLETS,
+  payload: { multisigWallets }
+});
+const _selectMultisigWallet = walletId => ({
+  type: USER_SELECT_MULTISIG_WALLET,
+  payload: { walletId }
+});
+const _selectMultisigProposal = proposalId => ({
+  type: USER_SELECT_MULTISIG_PROPOSAL,
+  payload: { proposalId }
+});
+const _setMultisigWalletInfo = (walletId, multisigWalletInfo) => ({
+  type: SET_MULTISIG_WALLET_INFO,
+  payload: { walletId, multisigWalletInfo }
+});
+const _setMultisigProposal = (walletId, proposals) => ({
+  type: SET_MULTISIG_WALLET_PROPOSALS,
+  payload: { walletId, proposals }
+});
+const _setMultisigProposalMTX = (walletId, proposalId, mtx) => ({
+  type: SET_MULTISIG_PROPOSAL_MTX,
+  payload: { walletId, proposalId, mtx }
+});
+const _selectXPUB = xpub => ({
+  type: USER_SELECT_XPUB,
+  payload: { selectedXPUB: xpub }
+});
+
+export const selectXPUB = xpub => async dispatch => {
```

Comment:
Why do these need to be split up with a `_` version and a normal with dispatch?

Reply:
I thought it was the redux was to have functions that only return objects that get invoked inside of dispatch, so I split them all out. They aren't really being reused and I think it would be a lot cleaner if they were not all apart like how they are. Ok with me doing that?

Reply:
oh, ok I see what's going on. You're using the async dispatcher convention for some actions that are not async. For normal action creators, the dispatch is done in the component via the `mapDispatchToProps` and redux' `connect`. There's some more information on connect [here](https://redux.js.org/basics/usage-with-react#implementing-container-components) and on async action creators with redux thunk [here](https://redux.js.org/advanced/async-actions#async-action-creators). I'm not sure it makes sense to necessarily use the `_` convention since the plain action creator could still be used elsewhere if you've already got the payload.